superset-frontend/src/SqlLab/components/TableElement/index.tsx

```diff
@@ -221,12 +230,15 @@ const TableElement = ({ table, actions, ...props }: TableElementProps) => {
     >
       <Tooltip
         id="copy-to-clipboard-tooltip"
-        placement="topLeft"
         style={{ cursor: 'pointer' }}
         title={table.name}
-        trigger={['hover']}
+        trigger={tableNameOverflow ? ['hover'] : []}
```

Comment:
I feel like if you're reading the property of a ref here to determine the trigger, you could prob just define the trigger as a const in this renderHeader function. I don't think you necessarily need the useEffect or state for this. That could simplify this code a bit. wdyt?

Reply:
Hello! I had to double check my knowledge on refs for a moment and also try this out because it would be awesome if I did not have to use the useEffect but unfortunately, it isn't working for me. 

If I understand React and refs correctly the ref only becomes defined after the element that it is attached to renders. So if I were to check if the width is longer than the container and put it in a const React would just say null < null and the ternary would go to the false value before the component even fully renders. Since there is nothing causing a change of state on that component, React won't rerender it again. 

The useEffect will wait until the component renders and then run once. We need this because the ref won't exist or have a length until it is rendered. Then we update the state of the conditional and then react will rerender any element using the state and the conditional will remove the hover trigger. 

My apologies if I am not explaining this well enough or if I completely misunderstood what you are suggesting

Reply:
Ohhh, my bad! It wasn't working for me because I left out the check for the element not being null and just had the comparison in the const. For some reason that doesn't work? Yes, it triggers a rerender, and the ref is updated and the comparison is made every time so it works exactly as expected. Very cool!

I agree this makes the code a lot simpler and more efficient. I also couldn't find anything wrong with doing it this way. I will change it and push up those changes shortly. Thanks!